### ISSUE TITLE
Improve documentation of numpy interface

### ISSUE THREAD

**Issue by cvjjm** [PennyLaneAI/pennylane#2175]
### Feature details

There have recently been several changes to the numpy/autograd interface that significantly impact how trainability of parameters is treated and which input and output types QNodes accept/produce with this interface.

At the same time I find the [documentation of this important interface](https://pennylane.readthedocs.io/en/stable/introduction/interfaces/numpy.html) is quite hard to find. The only way I found was to go via "Gradients and training", scroll down to "Training and interfaces" and then click on the "numpy" link hidden in the text. 

Also searching for "tensor" via the search box only produces hits under `pennylane.operation.Tensor` but does tell the user nothing about the `pennylane.numpy.tensor` objects returned from QNodes. Similarly, searching for `requires_grad` only shows the `pennylane.math.requires_grad` function but not hit concerning the argument of `pennylane.numpy.array()` with the same name, that is arguably more "user facing". 

It would be great if the peculiarities of the `pennylane.numpy` and `math` modules (the first does not even show up as a module in the menu under API) were explained in more detail and if this documentation were more accessible. Thanks for considering this!

### How important would you say this feature is?

2: Somewhat important. Needed this quarter.

**Comment by josh146** [comment]
@cvjjm this is a good idea indeed

**Comment by josh146** [comment]
@cvjjm I've added a PR to make the NumPy module (and the `tensor` class) visible in the docs in #2179. We'll also think on how to improve access/findability of the 'Autograd' page
